draco/analysis/ringmapmaker.py

```diff
@@ -223,7 +234,8 @@ def process(self, gstream):
 
         gsv = gstream.vis[:].local_array
         gsw = gstream.weight[:].local_array
-        gsr = gstream.redundancy[:]
+        if self.weight == "natural":
+            gsr = gstream.redundancy[:]
```

Comment:
Because the redundancy dataset is no longer guaranteed to exist, think there should be an extra check that either switches to a different weighting method or raises an informative error (probably the better option).